transaction-context/src/lib.rs

```diff
@@ -503,6 +511,33 @@ pub struct InstructionFrame {
     instruction_data: Vec<u8>,
 }
 
+/// This is a deconstructed InstructionFrame used to build a CompiledInstruction later
+pub struct DeconstructedInstruction {
+    /// Index of program account in transaction
+    pub program_account: u8,
+    /// Index of each instruction account in transaction
+    pub instruction_accounts: Vec<u8>,
+    /// Instruction data
+    pub data: Vec<u8>,
+    /// How many Instructions were on the stack after this one was pushed
+    pub stack_height: u8,
+}
```

Comment:
This is supposed to address https://github.com/anza-xyz/agave/pull/8213#discussion_r2391837926. I found it cumbersome to have a third data structure to mock CompiledInstructions, while we could just import it from Solana-message, whose component `CompiledInstruction` is shared with programs.

Reply:
Hm, since `TransactionContext::instruction_trace` is a private field, I think we could just make all fields in `InstructionFrame` public and thus avoid this new structure without actually exposing anything.

Reply:
Why are we overcomplicating while we can just import CompiledInstruction? If the import is a concern, we can remove it out from Cargo toml using a specific `cfg`.

Reply:
> Hm, since `TransactionContext::instruction_trace` is a private field, I think we could just make all fields in `InstructionFrame` public and thus avoid this new structure without actually exposing anything.

As I mentioned [here](https://github.com/anza-xyz/agave/pull/8292#issuecomment-3357449646), we may be able to remove `InstructionContext`, in which case, we would continue to be sharing `InstructionContext`.

Reply:
I accepted your suggestion and force pushed the fix. I didn't add a new commit, since the PR was still a draft.